app/kumactl/pkg/resources/inspect_envoy_proxy_client.go

```diff
@@ -34,22 +36,30 @@ type httpInspectEnvoyProxyClient struct {
 var _ InspectEnvoyProxyClient = &httpInspectEnvoyProxyClient{}
 
 func (h *httpInspectEnvoyProxyClient) ConfigDump(ctx context.Context, rk core_model.ResourceKey) ([]byte, error) {
-	return h.executeInspectRequest(ctx, rk, "xds")
+	return h.executeInspectRequest(ctx, rk, "xds", nil)
 }
 
 func (h *httpInspectEnvoyProxyClient) Stats(ctx context.Context, rk core_model.ResourceKey) ([]byte, error) {
-	return h.executeInspectRequest(ctx, rk, "stats")
+	return h.executeInspectRequest(ctx, rk, "stats", nil)
 }
 
 func (h *httpInspectEnvoyProxyClient) Clusters(ctx context.Context, rk core_model.ResourceKey) ([]byte, error) {
-	return h.executeInspectRequest(ctx, rk, "clusters")
+	return h.executeInspectRequest(ctx, rk, "clusters", nil)
 }
 
-func (h *httpInspectEnvoyProxyClient) executeInspectRequest(ctx context.Context, rk core_model.ResourceKey, inspectionPath string) ([]byte, error) {
+func (h *httpInspectEnvoyProxyClient) Config(ctx context.Context, rk core_model.ResourceKey, shadow bool, include []string) ([]byte, error) {
+	return h.executeInspectRequest(ctx, rk, "_config", url.Values{
+		"shadow":  {strconv.FormatBool(shadow)},
+		"include": include,
+	})
+}
+
+func (h *httpInspectEnvoyProxyClient) executeInspectRequest(ctx context.Context, rk core_model.ResourceKey, inspectionPath string, queryParams url.Values) ([]byte, error) {
 	resUrl, err := h.buildURL(rk, inspectionPath)
 	if err != nil {
 		return nil, errors.Wrap(err, "could not construct the url")
 	}
+	resUrl.RawQuery = queryParams.Encode()
```

Comment:
am I missing something or are we going to call `.Encode()` on `nil` from lines [47](https://github.com/kumahq/kuma/pull/9887/files#diff-f767b855e17c699f1100e5f7bd6f7bed1d7b7d683f7746c8cd1bb26fc2c970f3R47), 43, 39?

Reply:
There is a check inside so I was thinking it's okay https://github.com/golang/go/blob/master/src/net/url/url.go#L973. But we can check on the Kuma side as well

Reply:
or maybe it's better to pass empty `url.Values` instead

Reply:
just something I noticed if it won't NPE then I guess either one is good. Side q: do other builtin structures also have this? what's the idiomatic go way of doing this?

Reply:
not sure about the idiomatic way, but I think passing an empty `url.Values{}` is better than `nil` because that way code-readers don't have to check `.Encode()` implementation every time to ensure it's NPE safe.